### PR TITLE
Enable MGTwoLevelTransfer for non-precompiled p-coarsening sequences

### PR DESCRIPTION
**Pull request by peterrum**
References https://github.com/dealii/dealii/pull/10417#discussion_r492534103.

@kronbichler I guess this is what you were thinking about?

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]
This looks good, yes.

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 1]
```diff
@@ -100,19 +111,22 @@ namespace
...
+        const unsigned int degree_coarse_ = numbers::invalid_unsigned_int)
     {
       internal::FEEvaluationImplBasisChange<
         internal::evaluate_general,
         internal::EvaluatorQuantity::value,
         dim,
-        degree_coarse + 1,
-        degree_fine + 1,
+        degree_coarse == -1 ? 0 : (degree_coarse + 1),
+        degree_fine == -1 ? 0 : (degree_fine + 1),
```
[the remark this answers is not in the document]
Comment (peterrum) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h]: Done!

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 2]
```diff
@@ -92,15 +92,16 @@ class MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>
...
   /**
-   * Check if the polynomial transfer between @p fe_degree_fine and
-   * @p fe_degree_coarse is supported.
+   * Check if a fast templated version of the polynomial transfer between
+   * @p fe_degree_fine and @p fe_degree_coarse is available.
    *
    * @note Currently, the polynomial coarsening strategies: 1) go-to-one,
-   *   2) bisect, and 3) decrease-by-one is supported.
+   *   2) bisect, and 3) decrease-by-one are precompiled with templates for
+   *   degrees up to 9..
```
Comment (tjhei) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:100]:
```suggestion
   *   degrees up to 9.
```

**Review by tjhei — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]

**Pull request merged**